embassy-stm32/src/exti.rs

```diff
@@ -20,9 +44,9 @@ pub struct ExtiPin<T: Instance> {
 }
 
 impl<T: Instance> ExtiPin<T> {
-    pub fn new(mut pin: T, interrupt: T::Interrupt) -> Self {
+    pub fn new(mut pin: T, interrupt: T::Interrupt, syscfg: &mut SysCfg) -> Self {
         cortex_m::interrupt::free(|_| {
-            pin.make_source();
+            pin.make_source(syscfg);
```

Comment:
Do you still need this `interrupt::free` here ?

Reply:
`syscfg` is provided as a convenience. other MCUs may in the future transmute other items here. It could be removed for now.